cg/meta/workflow/nf_analysis.py

```diff
@@ -280,3 +292,59 @@ def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
                 )
             files.append(FileDeliverable(**file))
         return PipelineDeliverables(files=files)
+
+    def get_multiqc_json_path(self, case_id: str) -> Path:
+        """Return the path of the multiqc_data.json file."""
+        return Path(self.root_dir, case_id, "multiqc", "multiqc_data", "multiqc_data.json")
+
+    def get_multiqc_json_metrics(
+        self, case_id: str, pipeline_metrics: Optional[dict] = None
+    ) -> list[MetricsBase]:
+        """Get a multiqc_data.json file and returns metrics and values formatted."""
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        sample_id: str = case.links[0].sample.internal_id
+        multiqc_json: MultiqcDataJson = MultiqcDataJson(
+            **read_json(file_path=self.get_multiqc_json_path(case_id=case_id))
+        )
+        metrics_values: dict = {}
+        for key in multiqc_json.report_general_stats_data:
+            if case_id in key:
+                metrics_values.update(list(key.values())[0])
+                LOG.info(f"Key: {key}, Values: {list(key.values())[0]}")
+        return [
+            MetricsBase(
+                header=None,
+                id=sample_id,
+                input="multiqc_data.json",
+                name=metric_name,
+                step="multiqc",
+                value=metric_value,
+                condition=pipeline_metrics.get(metric_name, None),
+            )
+            for metric_name, metric_value in metrics_values.items()
+        ]
+
+    def validate_qc_metrics(self, case_id: str, dry_run: bool = False) -> None:
+        """Validate the information from a qc metrics deliverable file."""
+
+        if dry_run:
+            LOG.info("Dry-run: QC metrics validation would be performed")
+            return
+
+        LOG.info("Validating QC metrics")
+        try:
+            metrics_deliverables_path: Path = self.get_metrics_deliverables_path(case_id=case_id)
+            qc_metrics_raw: dict = ReadFile.get_content_from_file(
+                file_format=FileFormat.YAML, file_path=metrics_deliverables_path
+            )
+            MetricsDeliverablesCondition(**qc_metrics_raw)
```

Comment:
```suggestion
            MetricsDeliverablesCondition.model_validate(qc_metrics_raw)
```

Reply:
I commited this suggestion but I assume that this [line](
https://github.com/Clinical-Genomics/cg/blob/master/cg/models/deliverables/metric_deliverables.py#L4) has to change as well to:

`from pydantic import BaseModel, Field, validator, model_validator`

Reply:
Yes! `pydantic.v1` import have the become `pydantic` imports you have to check some things though. Between pydantic v1 and pydantic v2 there are some deprecating changes. For example, the `validator` function is deprecated and a different approach is needed to achieve the same functionality.

Reply:
@ChrOertlin I updated the PR but I get the following error:

`ERROR  - RuntimeError: no validator found for <class 'cg.models.mip.mip_metrics_deliverables.MIPParsedMetrics'>, see `arbitrary_types_allowed` in Config`

What would be the best way to proceed since it seems that MIP metrics are affected?

Reply:
MIP metrics deliverables would need to be updated and validated if I change this.
I will leave it as pydantic v.1 at the moment and those changes need to be migrated at a different PR. I am resolving this.